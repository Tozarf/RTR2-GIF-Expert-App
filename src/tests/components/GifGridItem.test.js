import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import React from 'react'

describe('Testing GifGriItem', () => {
  const title= "un titulo";
  const url="https://localhost/algo.jpg"
  // const animation= "animate__heartBeat"
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('should show the component correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
    
  test('should have a paragraph with the title', () => {
    
      const p = wrapper.find("p");
      
      expect(p.text().trim()).toBe(title)

  })

  test('img should have url and alt matching to the props', () => {
    
      const img= wrapper.find("img")
      // console.log(img.props())
      const {src, alt}= img.props()
      
      expect(src).toBe(url)
      expect(alt).toBe(title)

  })
  
  test('div should have the animate__heartBeat ', () => {
    
    const div= wrapper.find("div")
    const animation= "animate__heartBeat"
    const className = div.prop("className")
    

    expect(className.includes(animation)).toBe(true)  
    
  })
  
  
})
