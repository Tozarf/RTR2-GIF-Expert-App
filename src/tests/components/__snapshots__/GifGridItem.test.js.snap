// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGriItem should show the component correctly 1`] = `
<div
  className="card animate__animated animate__heartBeat"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
